packages/webapp/middleware.ts

```diff
@@ -0,0 +1,48 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+const trafficPercentage = 10;
+
+const cfCookieName = 'cf-override';
+
+const isDev = process.env.NODE_ENV !== 'production';
+
+const middleware = async (req: NextRequest): Promise<NextResponse> => {
```

Comment:
adding middleware would incur some costs on Vercel side:

> 1M invocations /month included
then starting at $0.65 per million

Reply:
how many invocations do we expect?